pkg/adapters/grpc/errors.go

```diff
@@ -15,25 +15,51 @@
 package grpc
 
 import (
+	"encoding/json"
+	"fmt"
+
 	"github.com/nitric-dev/membrane/pkg/plugins/errors"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
+type grpcError struct {
+	code    codes.Code `json:"-"`
+	Code    string     `json:"code"`
+	Msg     string     `json:"msg,omitempty"`
+	Service string     `json:"service,omitempty"`
+}
```

Comment:
Why not implement this using the standard richer error model guidance? https://grpc.io/docs/guides/error/#richer-error-model

Reply:
I have to admit I didn't really look at this. Will do now.

Reply:
Looks like we would define a standard protobuf error message type, and then use a protobuf builders to build the structure, and then use the underlying Status type to add this `WithDetails()`. 

Not sure how this would be expressed on the client side, possibly it would be deserialized as "metadata" in the Java GRPC library. Would need to give it a try.